src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -53,9 +59,84 @@ public FilterClause deserialize(
                 entry.getKey(), jsonNodeValue(entry.getKey(), entry.getValue())));
       }
     }
+
+    validate(expressionList);
     return new FilterClause(expressionList);
   }
 
+  private void validate(List<ComparisonExpression> expressionList) {
+    for (ComparisonExpression expression : expressionList) {
+      expression.filterOperations().forEach(operation -> validate(expression.path(), operation));
+    }
+  }
+
+  private void validate(String path, FilterOperation<?> filterOperation) {
+    if (filterOperation.operator() instanceof ValueComparisonOperator valueComparisonOperator) {
+      switch (valueComparisonOperator) {
+        case IN -> {
+          if (!path.equals(DocumentConstants.Fields.DOC_ID)) {
+            throw new JsonApiException(
+                ErrorCode.INVALID_FILTER_EXPRESSION, "Can use $in operator only on _id field");
+          }
+
+          if (filterOperation.operand().value() instanceof List<?> list) {
+            if (list.isEmpty()) {
+              throw new JsonApiException(
+                  ErrorCode.INVALID_FILTER_EXPRESSION, "$in operator must have at least one value");
+            }
+          } else {
+            throw new JsonApiException(
+                ErrorCode.INVALID_FILTER_EXPRESSION, "$in operator must have array");
```

Comment:
nit: node type is written as "ARRAY" so maybe use that instead of lower case.

Also: would be nice to include actual type but I guess that's difficult since it's converted from JSON to Java Object so can't easily display that?

Reply:
Document ID can be any of text, number, boolean and null. Would be like a long error. 
Change the casing of array as suggested.